keras/models.py

```diff
@@ -87,7 +87,11 @@ def compile(self, optimizer, loss, class_mode="categorical", theano_mode=None):
             raise Exception("Invalid class mode:" + str(class_mode))
         self.class_mode = class_mode
 
-        updates = self.optimizer.get_updates(self.params, self.regularizers, self.constraints, train_loss)
+        if hasattr(self, 'cost_updates'):
+            for u in self.cost_updates:
+                train_loss += u()
```

Comment:
Yes, that would be a good idea. We would need to monitor accuracy and speed along the way. Incorporating regularization in the loss should result in higher accuracy but *might* cause performance issues.